src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/readers/RecursiveJSONParser.java

```diff
@@ -127,6 +131,26 @@ private ObjectNode expandIncludesInObject(ObjectNode input) throws IOException {
             result = processMergeDirective(result);
         }
 
+        // iterate here again to put DYNAMIC_INCLUDE_FLAT_KEY as a parent node
```

Comment:
This method has become too long. Is better to refact it a bit i.e. by putting this code into a separate method.

Reply:
removed as you suggested in your other comment since we do not have to identify 2 different nodes that as includeFlat keyword